pubsub_test.go, TestConnectionEvents fail

Link to [the failed job](https://github.com/eclipse-kanto/suite-connector/actions/runs/3344546141/jobs/5540171122).
```
2022-10-28T11:34:51.8162211Z tests_1      | === RUN   TestConnectionEvents
2022-10-28T11:34:51.8170673Z tests_1      |     pubsub_test.go:717: 
2022-10-28T11:34:51.8171099Z tests_1      |         	Error Trace:	pubsub_test.go:717
2022-10-28T11:34:51.8171555Z tests_1      |         	Error:      	Should be true
2022-10-28T11:34:51.8172213Z tests_1      |         	Test:       	TestConnectionEvents
2022-10-28T11:34:51.8177765Z tests_1      |     pubsub_test.go:719: 
2022-10-28T11:34:51.8180530Z tests_1      |         	Error Trace:	pubsub_test.go:719
2022-10-28T11:34:51.8183306Z tests_1      |         	Error:      	Should be false
2022-10-28T11:34:51.8186166Z tests_1      |         	Test:       	TestConnectionEvents
2022-10-28T11:34:51.8186939Z tests_1      | --- FAIL: TestConnectionEvents (0.00s)
```

--- a/connector/pubsub_test.go
+++ b/connector/pubsub_test.go
@@ -679,11 +679,16 @@ func TestSimpleDelivery(t *testing.T) {
 }
 
 type connectionEventListener struct {
-	events chan<- bool
+	connectEvents    chan<- bool
+	disconnectEvents chan<- bool
 }
 
 func (h *connectionEventListener) Connected(connected bool, err error) {
-	h.events <- connected
+	if connected {
+		h.connectEvents <- connected
+	} else {
+		h.disconnectEvents <- connected
+	}
 }
 
 func TestConnectionEvents(t *testing.T) {
@@ -695,18 +700,22 @@ func TestConnectionEvents(t *testing.T) {
 	client, err := conn.NewMQTTConnection(config, watermill.NewShortUUID(), nil)
 	require.NoError(t, err)
 
-	events := make(chan bool, 8)
+	connectEvents := make(chan bool, 2)
+	disconnectEvents := make(chan bool, 2)
+
 	l1 := &connectionEventListener{
-		events: events,
+		connectEvents:    connectEvents,
+		disconnectEvents: disconnectEvents,
 	}
 	client.AddConnectionListener(l1)
+	client.AddConnectionListener(l1)
+	client.RemoveConnectionListener(l1)
 
 	l2 := &connectionEventListener{
-		events: events,
+		connectEvents:    connectEvents,
+		disconnectEvents: disconnectEvents,
 	}
 	client.AddConnectionListener(l2)
-	client.AddConnectionListener(l2)
-	client.RemoveConnectionListener(l2)
 
 	future := client.Connect()
 	<-future.Done()
@@ -714,9 +723,9 @@ func TestConnectionEvents(t *testing.T) {
 	client.Disconnect()
 
 	//Check for connected event
-	assert.True(t, <-events)
+	assert.True(t, <-connectEvents)
 	//Check for disconnected event
-	assert.False(t, <-events)
+	assert.False(t, <-disconnectEvents)
 }
 
 func TestMarshalError(t *testing.T) {

